tokio/src/util/linked_list.rs

```diff
@@ -174,6 +174,21 @@ impl<L: Link> LinkedList<L, L::Target> {
         true
     }
 
+    // Counts the elements of the LinkedList
+    pub(crate) fn count(&self) -> usize {
+        let mut count = 0;
+        let mut current = self.head;
+
+        while let Some(node) = current {
+            unsafe {
+                count += 1;
+                current = L::pointers(node).as_ref().get_next()
+            }
+        }
+
+        count
+    }
```

Comment:
This makes the `active_task_count` call rather expensive compared to other metrics. It might be better to add a counter.

Reply:
It's true, also we are locking the list. I can give it a shot.

Reply:
Yes, I would add an additional task count atomic and inc / dec it as needed.